mobile/native/src/health.rs

```diff
@@ -22,44 +28,96 @@ pub enum ServiceStatus {
 
 pub type ServiceUpdate = (Service, ServiceStatus);
 
+/// Senders for the health status updates.
+///
+/// Meant to be injected into the services that need to publish their health status.
 pub struct Tx {
     pub orderbook: watch::Sender<ServiceStatus>,
 }
 
-/// Entity that gathers all the health data and sends notifications
+/// Entity that gathers all the service health data and publishes notifications
 pub struct Health {
-    _monitoring_task: tokio::task::JoinHandle<()>,
+    _tasks: Vec<tokio::task::JoinHandle<()>>,
 }
 
 impl Health {
-    pub fn new(runtime: &Runtime) -> (Self, Tx) {
-        let (orderbook_tx, mut orderbook_rx) = watch::channel(ServiceStatus::Unknown);
-
-        let _monitoring_task = runtime.spawn(async move {
-            loop {
-                select! {
-                    result = orderbook_rx.changed() => {
-                        match result {
-                            Ok(()) => {
-                                let status = orderbook_rx.borrow();
-                                event::publish(&EventInternal::ServiceHealthUpdate((Service::Orderbook, *status)));
-                            }
-                            Err(_) => {
-                                tracing::error!("Sender dropped");
-                                event::publish(&EventInternal::ServiceHealthUpdate((Service::Orderbook, ServiceStatus::Unknown)));
-                                break;
-                            },
-                        }
-                    },
-                }
-            }
-        });
+    pub fn new(config: Config, runtime: &Runtime) -> (Self, Tx) {
+        let (orderbook_tx, orderbook_rx) = watch::channel(ServiceStatus::Unknown);
+
+        let config: ConfigInternal = config.into();
+
+        let mut tasks = Vec::new();
+
+        let orderbook_monitoring =
+            runtime.spawn(publish_status_updates(Service::Orderbook, orderbook_rx));
+        tasks.push(orderbook_monitoring);
+
+        let (coordinator_tx, coordinator_rx) = watch::channel(ServiceStatus::Unknown);
+
+        let coordinator_health_endpoint =
+            "http://".to_owned() + &config.http_endpoint.to_string() + "/health";
+        let check_coordinator = tokio::spawn(check_health_endpoint(
+            coordinator_health_endpoint,
+            coordinator_tx,
+        ));
+        tasks.push(check_coordinator);
+        let coordinator_monitoring =
+            runtime.spawn(publish_status_updates(Service::Coordinator, coordinator_rx));
+        tasks.push(coordinator_monitoring);
 
         (
-            Self { _monitoring_task },
+            Self { _tasks: tasks },
             Tx {
                 orderbook: orderbook_tx,
             },
         )
     }
 }
+
+/// Publishes the health status updates for a given service to the event hub
+async fn publish_status_updates(service: Service, mut rx: watch::Receiver<ServiceStatus>) {
+    loop {
+        select! {
```

Comment:
🔧 We don't need a `select` if we are only polling one future.

I suppose we can keep it if we know we are gonna use it for something else soon, but I'd rather keep the code simple until we need to make it complicated.

Reply:
I don't think using `select!` is wrong in any way - it's not less performant solution, and it's very clear what the code does....

Reply:
You can just `await` the single future as we are not waiting for multiple concurrent branches, which is what `tokio::select` is for.

Reply:
makes sense!